lib/config-validator.js

```diff
@@ -95,17 +100,52 @@ function validateRuleOptions(id, options, source) {
 }
 
 /**
+ * Validates an environment object
+ * @param {object} environment The environment config object to validate.
+ * @param {string} source The location to report with any errors.
+ * @returns {void}
+ */
+function validateEnvironment(environment, source) {
+
+    // undefined environments are ok
+    if (environment === "undefined" || environment == null) {
```

Comment:
Are you meaning to do this?

```
if (!environment)
```

or 

```js
if (environment === undefined || environment == null)
```

I'm just not clear on how `environment` could end up equal to the string `"undefined"`.